### PR TITLE
Convert elastic tests to use tensor mechanics

### PR DESCRIPTION
**Pull request by hchen139**
refs #8642

### CODE REVIEW

File: modules/combined/tests/check_error/contact_displacements.i [hunk 1]
```diff
@@ -1,37 +1,23 @@
...
-[] # Variables
-
-[SolidMechanics]
-  [./solid]
-    disp_x = disp_x
-    disp_y = disp_y
-    disp_z = disp_z
+[Kernels]
+  [./TensorMechanics]
+    displacements = 'disp_x disp_y disp_z'
```
Comment (dschwen) [inline modules/combined/tests/check_error/contact_displacements.i]: This is already in GlobalParams!

File: modules/combined/tests/check_error/contact_displacements.i [hunk 2]
```diff
@@ -63,32 +48,24 @@
...
+    type = ComputeIsotropicElasticityTensor
     youngs_modulus = 1e6
     poissons_ratio = 0.0
-
   [../]
-
-[] # Materials
+  [./strain]
+    type = ComputeSmallStrain
+    displacements = 'disp_x disp_y disp_z'
```
Comment (dschwen) [inline modules/combined/tests/check_error/contact_displacements.i]: ditto - please remove all those

File: modules/combined/tests/check_error/contact_displacements_sm.i [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    value = 0.0
+  [../]
+[]
+
+[Contact]
+  [./fred]
+    master = 1
+    slave = 2
+    disp_x = disp_x
+    disp_y = disp_y
```
Comment (dschwen) [inline modules/combined/tests/check_error/contact_displacements_sm.i]: This is deprecated syntax now and will fail the tests. Pleas put a `displacements = ` into GlobalParams and remove the `disp_[xyz]` from Contact, Pressure, and Density everywhere.

**Review by dschwen — CHANGES_REQUESTED** [review]

File: modules/combined/tests/check_error/contact_displacements.i (hunk 1 above)
Comment (hchen139) [inline modules/combined/tests/check_error/contact_displacements.i]: The test will fail with "unable to find nearest node!" if put the displacements in the GlobalParams block.

File: modules/tensor_mechanics/src/materials/ComputeIsotropicElasticityTensor.C [hunk 4]
```diff
@@ -34,6 +34,26 @@ ComputeIsotropicElasticityTensor::ComputeIsotropicElasticityTensor(
     _shear_modulus(_shear_modulus_set ? getParam<Real>("shear_modulus") : -1),
     _youngs_modulus(_youngs_modulus_set ? getParam<Real>("youngs_modulus") : -1)
 {
+  int num_elastic_constants = _bulk_modulus_set + _lambda_set + _poissons_ratio_set +
```
Comment (dschwen) [inline modules/tensor_mechanics/src/materials/ComputeIsotropicElasticityTensor.C]: should be `unsigned int` unless you expect a negative number of elastic constants...

File: modules/combined/tests/check_error/contact_displacements.i (hunk 1 above)
Comment (hchen139) [inline modules/combined/tests/check_error/contact_displacements.i]: I looked at the ContactAction.C, and turned out that the intend of this test is to check whether the displacements is set in the mesh block.

File: modules/tensor_mechanics/src/materials/ComputeIsotropicElasticityTensor.C (hunk 4 above)
Comment (hchen139) [inline modules/tensor_mechanics/src/materials/ComputeIsotropicElasticityTensor.C]: will update this.

**Comment by hchen139** [comment]
@bwspenc do you have an idea why the xfem init_solution_propagation test failed?

**Comment by bwspenc** [comment]
Ugh!  I just added that test.  It passed in my initial PR, and I assume it would have run on the same platforms.  I'll look into that.  Maybe I need to skip it or something.

File: modules/combined/tests/check_error/contact_displacements.i (hunk 1 above)
Comment (dschwen) [inline modules/combined/tests/check_error/contact_displacements.i]: Ok, I see that now.

**Review by dschwen — APPROVED** [review]

**Pull request merged**